### PR TITLE
Alerting: Update migration to put alerts to the default folder if dashboard folder is missing

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What is this feature?**
Grafana database is designed a way that constraints are not enforced on the database level (no foreign keys). This may cause inconsistencies with data. During the migration of alert rules to unified alerting, the migration code tries to migrate the rules to the same folder where the dashboard they are extracted from is stored. However, sometimes, due to inconsistencies of the database, the folder can be missing, and this causes the migration to fail.
This PR changes the migration to not fail. Instead, migration will emit a warning message `Failed to find folder for dashboard. Migrate rule to the default folder` and will try to put the alert to the folder `General Alerting`. 

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.
- [x] There are no known compatibility issues with older supported versions of Grafana, or plugins.
- [x] It passes the [Hosted Grafana feature readiness review](https://docs.google.com/document/d/1QL9Ly8KnXzpb6ISbg49pTODRO5mhA5tkkfIZVX6pqQU/edit) for observability, scalability, performance, and security.

### CODE REVIEW

File: pkg/services/sqlstore/migrations/ualert/ualert.go [hunk 1]
```diff
@@ -335,29 +355,20 @@ func (m *migration) Exec(sess *xorm.Session, mg *migrator.Migrator) error {
 			// get folder if exists
 			f, err := folderHelper.getFolder(dash, da)
 			if err != nil {
-				return MigrationError{
-					Err:     err,
-					AlertId: da.Id,
+				// if folder does not exist then the dashboard is an orphan and therefore rules can be skipped
```
Comment (JacobsonMT) [inline pkg/services/sqlstore/migrations/ualert/ualert.go:358]: NIT: I think this comment describes the opposite of what is happening here, maybe change to:
```suggestion
				// If folder does not exist then the dashboard is an orphan and we migrate the alert to the general folder.
```

**Review by JacobsonMT — APPROVED** [review]
LGTM, one NIT for the comment but I'l approve as is

**Review by armandgrillet — APPROVED** [review]
LGTM

**Comment by yuri-tceretian** [comment]
We decided to backport this fix down to 8.4.x

**Pull request merged**